src/components/FindMovie/FindMovie.js

```diff
@@ -1,55 +1,137 @@
-import movies from '../../api/movies.json';
-
-export const FindMovie = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+import { getFilm } from '../../api/api';
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+const imdbUrl = 'https://www.imdb.com/title/';
+
+export const FindMovie = ({ addMovie }) => {
+  const [query, setQuery] = useState('');
+  const [errorVisibility, seterrorVisibility] = useState(false);
+  const [filmsInList, setFilmInList] = useState([
+    'tt1375666', 'tt0314331', 'tt0319262', 'tt3748528', 'tt0457939',
+  ]);
+  const [film, setFilm] = useState({
+    title: null,
+    description: null,
+    imgUrl: null,
+    imdbUrl: null,
+    imdbId: null,
+    response: false,
+  });
+
+  useEffect(() => {
+    if (errorVisibility) {
+      seterrorVisibility(false);
+    }
+
+    document.querySelector('.input').classList.remove('is-danger');
+  }, [query]);
+
+  const hadleClickAddToList = () => {
+    setQuery('');
+
+    if (!film.response) {
+      return;
+    }
+
+    if (filmsInList.includes(film.imdbId)) {
+      return;
+    }
 
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
+    setFilmInList([...filmsInList, film.imdbId]);
+    addMovie(film);
+  };
+
+  const handleSubmitFindMovie = async(event) => {
+    if (query === '') {
+      return;
+    }
+
+    event.preventDefault();
+
+    const movie = await getFilm(query);
+    const { Title, Plot, Poster, imdbID, Response } = movie;
+
+    if (Response === 'True') {
+      setFilm({
+        title: Title,
+        description: Plot,
+        imgUrl: Poster,
+        imdbId: imdbID,
+        imdbUrl: imdbUrl + imdbID,
+        response: Response === 'True',
+      });
+    } else {
+      setFilm({});
+      seterrorVisibility(true);
+      document.querySelector('input').classList.add('is-danger');
+    }
+  };
+
+  return (
+    <>
+      <form
+        className="find-movie"
+        onSubmit={handleSubmitFindMovie}
+      >
+        <div className="field">
+          <label className="label" htmlFor="movie-title">
+            Movie title
+          </label>
+
+          <div className="control">
+            <input
+              type="text"
+              id="movie-title"
+              placeholder="Enter a title to search"
+              className="input"
+              value={query}
+              onChange={event => setQuery(event.target.value)}
```

Comment:
```suggestion
              onChange={
                (event) => setQuery(event.target.value);
                seterrorVisibility(false);
              }
```
As a variant, you can reset an error here onChange fand not use `useEffect`